security/advancedtls/crl_test.go

```diff
@@ -636,17 +641,33 @@ func setupTLSConn(t *testing.T) (net.Listener, *x509.Certificate, *ecdsa.Private
 // TestVerifyConnection will setup a client/server connection and check revocation in the real TLS dialer
 func TestVerifyConnection(t *testing.T) {
 	l, cert, key := setupTLSConn(t)
+
+	done := make(chan interface{}, 1)
+	var wg sync.WaitGroup
+	wg.Add(1)
 	go func() {
+	ListenLoop:
 		for {
 			conn, err := l.Accept()
 			if err != nil {
-				t.Errorf("tls.Accept failed err = %v", err)
+				switch {
+				case <-done:
+					wg.Done()
+					break ListenLoop
+				default:
+					t.Errorf("tls.Accept failed err = %v", err)
```

Comment:
In this case, I think you'll start busy looping on `Accept` calls, so you still need to `return`.  And you'll need to `Done` the waitgroup / close the channel.  I'd recommend something more like this:

```go
go func() {
	defer close(exit)
	for {
		conn, err := l.Accept()
		if err != nil {
			switch {
			case <-done:
			default:
				t.Errorf(...)
			}
			return
		}
		// conn.Write + Close
	}
}()
```